src/spikeinterface/postprocessing/correlograms.py

```diff
@@ -595,3 +604,418 @@ def _compute_correlograms_one_segment_numba(
+    assert winsize_bins >= 1, "Number of bins must be >= 1"
+    assert winsize_bins % 2 == 1, "Number of bins must be odd"
+    bin_times_ms = np.linspace(-win_size / 2, win_size / 2, num=winsize_bins)
+
+    if firing_rate_quantiles is not None:
+        num_firing_rate_quantiles = len(firing_rate_quantiles)
+    spike_counts = np.zeros(
+        (num_firing_rate_quantiles, len(bin_times_ms))
+    )  # Counts number of occurences of spikes in a given bin in time axis
+    firing_rate_bin_occurence = np.zeros(num_firing_rate_quantiles, dtype=np.int64)  # total occurence
+
+    # Samples per bin
+    samples_per_bin = int(np.ceil(fs / (1000 / bin_size)))
+
+    num_segments = sorting.get_num_segments()
+    # Convert times_1 and times_2 (which are in units of fs to units of bin_size)
+    for segment_index in range(num_segments):
+        spike_times = sorting.get_unit_spike_train(unit_id, segment_index=segment_index)
+
+        # Convert to bin indices
+        spike_bins = np.floor(spike_times / samples_per_bin).astype(np.int64)
+
+        if len(spike_bins) <= 1:
+            continue  # Need at least 2 spikes for ACG
+
+        # Create a binary spike train spanning the entire time range
+        max_bin = int(np.ceil(spike_bins[-1] + 1))
+        spiketrain = np.zeros(max_bin + winsize_bins, dtype=bool)  # Add extra space for window
+        spiketrain[spike_bins] = True
+
+        # Convert spikes to firing rate using the inverse ISI method
+        firing_rate = np.zeros(max_bin)
+        for i in range(1, len(spike_bins) - 1):
+            start = 0 if i == 0 else (spike_bins[i - 1] + (spike_bins[i] - spike_bins[i - 1]) // 2)
+            stop = max_bin if i == len(spike_bins) - 1 else (spike_bins[i] + (spike_bins[i + 1] - spike_bins[i]) // 2)
+            current_firing_rate = 1.0 / ((stop - start) * (bin_size / 1000))
+            firing_rate[start:stop] = current_firing_rate
+
+        # Smooth the firing rate using numpy convolution function if requested
+        if isinstance(smoothing_factor, (int, float)) and smoothing_factor > 0:
+            kernel_size = int(np.ceil(smoothing_factor / bin_size))
+            half_kernel_size = kernel_size // 2
+            kernel = np.ones(kernel_size) / kernel_size
+            smoothed_firing_rate = np.convolve(firing_rate, kernel, mode="same")
+
+            # Correct manually for possible artefacts at the edges
+            for i in range(kernel_size):
+                start = max(0, i - half_kernel_size)
+                stop = min(len(firing_rate), i + half_kernel_size)
+                smoothed_firing_rate[i] = np.mean(firing_rate[start:stop])
+            for i in range(len(firing_rate) - kernel_size, len(firing_rate)):
+                start = max(0, i - half_kernel_size)
+                stop = min(len(firing_rate), i + half_kernel_size)
+
+                smoothed_firing_rate[i] = np.mean(firing_rate[start:stop])
```

Comment:
OK with this but this prevent using other kernel for smoothing.
Is this border handling usefull ?